staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -91,6 +92,23 @@ type Config struct {
 	StorageObjectCountTracker flowcontrolrequest.StorageObjectCountTracker
 }
 
+// ConfigForResource is a Config specialized to a particular `schema.GroupResource`
+type ConfigForResource struct {
+	// Config is the resource-independent configuration
+	Config
+
+	// GroupResourceString is the `String()` of the `schema.GroupResource`
+	GroupResourceString string
+}
+
+// ForResource specializes to the given resource
+func (config Config) ForResource(resource schema.GroupResource) *ConfigForResource {
```

Comment:
Let's make this method on `(config *Config)` instead

Reply:
Why?

Reply:
I don't think that output really matter.
By convention we're generally add methods on pointers.

Reply:
Actually, the type `ConfigForResource` says that it holds a copy of, rather than a reference to, the `Config`.  So I do not need to insist on this function saying that.